src/platform/packages/private/kbn-grid-layout/grid/grid_row/grid_row_header.tsx

```diff
@@ -130,39 +189,41 @@ export const GridRowHeader = React.memo(
                 </EuiFlexItem>
                 {!readOnly && (
                   <>
-                    <EuiFlexItem grow={false}>
+                    {!isActive && (
+                      <EuiFlexItem grow={false}>
+                        <EuiButtonIcon
+                          iconType="trash"
+                          color="danger"
+                          className="kbnGridLayout--deleteRowIcon"
+                          onClick={confirmDeleteRow}
+                          aria-label={i18n.translate('kbnGridLayout.row.deleteRow', {
+                            defaultMessage: 'Delete section',
+                          })}
+                        />
+                      </EuiFlexItem>
+                    )}
+                    <EuiFlexItem
+                      grow={false}
+                      css={[styles.floatToRight, styles.visibleOnlyWhenCollapsed]}
+                    >
                       <EuiButtonIcon
-                        iconType="trash"
-                        color="danger"
-                        className="kbnGridLayout--deleteRowIcon"
-                        onClick={confirmDeleteRow}
-                        aria-label={i18n.translate('kbnGridLayout.row.deleteRow', {
-                          defaultMessage: 'Delete section',
-                        })}
-                      />
-                    </EuiFlexItem>
-                    <EuiFlexItem grow={false} css={[styles.hiddenOnCollapsed, styles.floatToRight]}>
-                      {/*
-                        This was added as a placeholder to get the desired UI here; however, since the
-                        functionality will be implemented in https://github.com/elastic/kibana/issues/190381
-                        and this button doesn't do anything yet, I'm choosing to hide it for now. I am keeping
-                        the `FlexItem` wrapper so that the UI still looks correct.
-                      */}
-                      {/* <EuiButtonIcon
                         iconType="move"
                         color="text"
                         className="kbnGridLayout--moveRowIcon"
                         aria-label={i18n.translate('kbnGridLayout.row.moveRow', {
                           defaultMessage: 'Move section',
                         })}
-                      /> */}
+                        onMouseDown={startInteraction}
```

Comment:
The whole code for touch interaction is written, but the action is not attached, I think this (and one more line for blocking the scrolling when using a touch, my next comment) will do the trick (tested)
```suggestion
                        onMouseDown={startInteraction}
                        onTouchStart={startInteraction}
```

Reply:
Woops!! Meant to do this and totally forgot 🙈 Thanks for catching that!! 13dbd36b67b6f7e63a9f846139baf09265e781ca